### PR TITLE
$queryRaw and $executeRaw GA

### PR DESCRIPTION
**Pull request by molebox**
This PR update the `$queryRaw` and `$executeRaw` docs to reflect the changes made in https://github.com/prisma/prisma/issues/7142

Fixes: #2157

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/058-transactions.mdx [hunk 1]
```diff
@@ -87,13 +87,22 @@ Instead of immediately awaiting the result of each operation when it's performed
 >
 > Refer to the 📖 [transactions guide](../../../guides/performance-and-optimization/prisma-client-transactions-guide#transaction-api) for more examples.
 
-### Interactive Transactions
+### Interactive Transactions (in Preview)
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/058-transactions.mdx]: Unrelated change to PR - should probably be merged into the epic branch so it does not show up here.

File: content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx [hunk 2]
```diff
@@ -823,11 +823,20 @@ It is now impossible for two people to book the same seat:
 3. Seat 3A claimed by Sorcha (`version` is incremented to 1, booking succeeds)
 4. Seat 3A claimed by Ellen (in-memory `version` (0) does not match database `version` (1) - booking does not succeed)
 
-### Interactive Transactions
+### Interactive Transactions (in Preview)
```
Comment (janpio) [inline content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx]: also unrelated change here

**Review by janpio — COMMENTED** [review]
- Line 16 can be adapted to list the new methods as well now
- SQL Injection section can probably be updated now with new methods

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 3]
```diff
@@ -111,19 +117,48 @@ result.forEach((x) => {
...
-> **Note**: The Prisma Client query engine standardizes the return type for all databases. Using the raw queries does not.
+> **Note**: The Prisma Client query engine standardizes the return type for all databases. **Using the raw queries does not**.
+
+## <inlinecode>$queryRawUnsafe</inlinecode>
+
+The `$queryRawUnsafe` method allows you to pass a raw string (or template string) to the database. 
+
+<Admonition type="warning">
+
+**By using this method you open up the possibility for SQL injection attacks**.
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]:
```suggestion
**By using this method you open up the possibility for SQL injection attacks.
We therefore strongly advise against using this method and use `queryRaw` instead, unless you have no alternative. If so, be sure to properly escape any variable to mitigate the risk of SQL injections.**
You can [learn more about SQL injections on Wikipedia](https://en.wikipedia.org/wiki/SQL_injection).
```

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 4]
```diff
@@ -30,15 +30,23 @@ Use cases for raw SQL include:
...
-const result = await prisma.$queryRaw('SELECT * FROM User;')
+const result = await prisma.$queryRaw`SELECT * FROM User`
 ```
 
-The same method is implemented as a [tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), which makes it easier to use [variables](#using-variables) by providing **placeholders**. The following example includes an `${email}` placeholder:
+The method is implemented as a [tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), which makes it easier to use [variables](#using-variables) by providing **placeholders**. The following example includes an `${email}` placeholder:
 
 ```ts no-lines
-const email = 'edna@prisma.io'
+const email = 'bruce@avengers.io'
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: Why switching to Avengers? I'd stick to something as neutral (and diverse) as possible ideally.

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 5]
```diff
@@ -47,8 +55,8 @@ const result =
 
 ```json5
 [
-  { id: 1, email: 'sarah@prisma.io', name: 'Sarah' },
-  { id: 2, email: 'alice@prisma.io', name: 'Alice' },
+  { id: 1, email: 'bruce@avengers.io', name: 'Bruce' },
+  { id: 2, email: 'tony@avengers.io', name: 'Tony' },
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: Cf. above. I'd try something more neutral and diverse.

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 6]
```diff
@@ -146,7 +181,38 @@ Be aware that:
...
+$executeRaw<T = unknown>(query: TemplateStringsArray | Prisma.Sql, ...values: any[]): PrismaPromise<T>;
+```
+
+## <inlinecode>$executeRawUnsafe</inlinecode>
+
+The `$executeRawUnsafe` method allows you to pass a raw string (or template string) to the database. Like `$executeRaw`, it does **not** return database records, but returns the number of rows affected.
+
+<Admonition type="warning">
+
+**By using this method you open up the possibility for SQL injection attacks**.
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]:
```suggestion
**By using this method you open up the possibility for SQL injection attacks.
We therefore strongly advise against using this method and use `executeRaw` instead, unless you have no alternative. If so, be sure to properly escape any variable to mitigate the risk of SQL injections.**
You can [learn more about SQL injections on Wikipedia](https://en.wikipedia.org/wiki/SQL_injection).
```

**Review by thebiglabasky — CHANGES_REQUESTED** [review]
Left some comments but generally looks good to me.
The interactive transactions content being there is a bit disturbing, but can work fine

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx (hunk 4 above)
Comment (molebox) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: Yeah, you are 100% correct. I had changed one and thought about changing all the others to be consistent. But its not really the place for it to be fair 😅

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 7]
```diff
@@ -111,19 +119,53 @@ result.forEach((x) => {
...
-> **Note**: The Prisma Client query engine standardizes the return type for all databases. Using the raw queries does not.
+> **Note**: The Prisma Client query engine standardizes the return type for all databases. **Using the raw queries does not**.
+
+## <inlinecode>$queryRawUnsafe</inlinecode>
+
+The `$queryRawUnsafe` method allows you to pass a raw string (or template string) to the database.
+
+<Admonition type="warning">
+
+By using this method you open up the possibility for SQL injection attacks. SQL injection attacks can expose your data, be it confidential or otherwise sensitive, to being deleted, modified, or even destroyed.
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]:
```suggestion
By using this method you open up the possibility for SQL injection attacks. SQL injection attacks can expose your data, be it confidential or otherwise sensitive, to being modified, or even destroyed.
```
Destroyed vs deleted didn't seem to make sense together?

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 8]
```diff
@@ -111,19 +119,53 @@ result.forEach((x) => {
...
+
+We strongly advice that you use the `$queryRaw` query instead. For more information on SQL injection attacks, see the [OWASP SQL Injection guide](https://www.owasp.org/index.php/SQL_Injection).
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: (Maybe that's my english 😄)
Don't we say "We strongly advise" with an 's'?

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 9]
```diff
@@ -136,7 +178,7 @@ const result: number =
 Be aware that:
 
 - `$executeRaw` does not support multiple queries in a single string (for example, `ALTER TABLE` and `CREATE TABLE` together).
-- Prisma Client submits prepared statements, and prepared statements only allow a subset of SQL statements (`START TRANSACTION`is not permitted). For example, ["MySQL: SQL Syntax Allowed in Prepared Statements"](https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html).
+- Prisma Client submits prepared statements, and prepared statements only allow a subset of SQL statements. For example, `START TRANSACTION` is not permitted. You can learn more about [“the syntax that MySQL allows in Prepared Statements here”](https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html).
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]:
```suggestion
- Prisma Client submits prepared statements, and prepared statements only allow a subset of SQL statements. For example, `START TRANSACTION` is not permitted. You can learn more about [the syntax that MySQL allows in Prepared Statements here](https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html).
```
Tiny detail :)

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 10]
```diff
@@ -146,7 +188,46 @@ Be aware that:
...
+$executeRaw<T = unknown>(query: TemplateStringsArray | Prisma.Sql, ...values: any[]): PrismaPromise<T>;
+```
+
+## <inlinecode>$executeRawUnsafe</inlinecode>
+
+The `$executeRawUnsafe` method allows you to pass a raw string (or template string) to the database. Like `$executeRaw`, it does **not** return database records, but returns the number of rows affected.
+
+<Admonition type="warning">
+
+By using this method you open up the possibility for SQL injection attacks. SQL injection attacks can expose your data, be it confidential or otherwise sensitive, to being deleted, modified, or even destroyed.
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]:
```suggestion
By using this method you open up the possibility for SQL injection attacks. SQL injection attacks can expose your data, be it confidential or otherwise sensitive, to being modified, or even destroyed.
```

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 11]
```diff
@@ -235,36 +318,45 @@ When you use `ILIKE`, the `%` wildcard character(s) should be included in the va
...
 
 PostgreSQL [does not support using `ALTER` in a prepared statement](https://www.postgresql.org/docs/current/sql-prepare.html), which means that the following queries will not work:
 
 ```ts
 await prisma.$executeRaw`ALTER USER prisma WITH PASSWORD '${password}'` // Tagged template
-await prisma.$executeRaw(`ALTER USER prisma WITH PASSWORD $1`, newPassword) // Parameterized query
+await prisma.$executeRaw`ALTER USER prisma WITH PASSWORD $1, newPassword` // Parameterized query
 ```
 
 You can use the following query to [work around the issue](https://github.com/prisma/prisma-client-js/issues/940#issuecomment-748014175), but be aware that this is potentially **unsafe** as `${password}` is not escaped:
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: I think there might be an issue here since we say $executeRaw is the safe one and $executeRawUnsafe is the unsafe one. How could $executeRaw be unsafe?

Comment (thebiglabasky) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: Shouldn't that then be done using the `Unsafe` signature?

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx [hunk 12]
```diff
@@ -235,36 +318,45 @@ When you use `ILIKE`, the `%` wildcard character(s) should be included in the va
...
+  ```ts
+  $queryRaw`` // Tagged template
+  $executeRaw`` // Tagged template
+  $queryRaw`query, variables` // Parameterized query
+  $executeRaw`query, variables` // Parameterized query
+  ```
+
+- Both versions of `$executeRaw` (tagged template and parameterized) can only run **one** query at a time. You cannot append a second query - for example, adding `DROP bobby_tables` to the end of an `ALTER`.
+
+If you cannot use tagged templates or parameterized queries (for example, you need to pass in table or column names as variables), you can instead use [`$queryRawUnsafe`](#query-raw-unsafe) or [`$executeRawUnsafe`](#execute-raw-unsafe) but **be aware that your code may be vulnerable to SQL injection**.
```
Comment (thebiglabasky) [inline content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: Note for later: it could be good to point people to how to properly escape strings etc. to prevent SQL injections whenever using Unsafe signature methods. This can be done in a separate effort, but would be worthwhile as a user, since you otherwise just see you're in this situation but left with the task to figure out how to deal with it on your own.

File: content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx [hunk 13]
```diff
@@ -823,7 +823,7 @@ It is now impossible for two people to book the same seat:
 3. Seat 3A claimed by Sorcha (`version` is incremented to 1, booking succeeds)
 4. Seat 3A claimed by Ellen (in-memory `version` (0) does not match database `version` (1) - booking does not succeed)
 
-### Interactive Transactions (in Preview)
+### Interactive Transactions
```
Comment (thebiglabasky) [inline content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx]: Still unrelated, but why dropping the (Preview) label here? In preparation for the GA of interactive transactions?

**Review by thebiglabasky — CHANGES_REQUESTED** [review]
I've spotted a couple of minor things.
One key question revolves around the ALTER table limitation and suggestion solution where I wonder if there is an error in the doc here.

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx (hunk 8 above)
Comment (molebox) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: AHHHHHHHH yes we do! I shall take your advice and change it!

ba-dum-tis

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx (hunk 11 above)
Comment (molebox) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx]: This section i wasnt sure about as it relates to an issue. I asked about this in slack yesterday too.

File: content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx (hunk 13 above)
Comment (molebox) [reply content/300-guides/100-performance-and-optimization/100-prisma-client-transactions-guide.mdx]: Yes, have i jumped the gun on that? Looking through my notes i can see that its not mentioned for 7th sept release so i think i may have made a mistake here.

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx (hunk 12 above)
Comment (molebox) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: I agree, a new section within this page about how to escape strings while using `$executeRawUnsafe` / `$queryRawUnsafe` would be ideal. Ive asked @millsp about that. Would be good to link to it from within the page

Comment (millsp) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: Somewhat related: I'm wondering if we could allow `ALTER`, `CREATE`, `DROP` in `$executeRaw` and instead of doing a prepared statement, we would automatically escape their inputs. I also think that we should look into providing some helper to escape input.

Comment (molebox) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: @millsp is this something you are thinking about for 7th sept release?

Comment (thebiglabasky) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: Those are interesting ideas indeed @millsp and I think we should track them in their individual issues, unless they would come with a breaking change, in such case that would be better to indeed do those before the new major.

**Review by thebiglabasky — APPROVED** [review]

File: content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx (hunk 12 above)
Comment (millsp) [reply content/200-concepts/100-components/02-prisma-client/090-raw-database-access.mdx:359]: That won't be a breaking change, so I'll keep the improvements for later.
